src/app/portlets/dot-edit-content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.spec.ts

```diff
@@ -126,17 +130,69 @@ describe('DotEditPageToolbarComponent', () => {
     });
 
     it('should hide the save button in live mode', () => {
-        component.stateSelected = PageMode.LIVE;
+        component.mode = PageMode.LIVE;
 
         fixture.detectChanges();
         const primaryAction: DebugElement = de.query(By.css('.edit-page-toolbar__save'));
 
         expect(primaryAction).toBeFalsy();
     });
 
-    it('should have lock page component', () => {
+    it('should have lock page component and no warn class', () => {
+        fixture.detectChanges();
         const lockSwitch: DebugElement = de.query(By.css('.edit-page-toolbar__locker'));
         expect(lockSwitch !== null).toEqual(true);
+        expect(lockSwitch.classes.warn).toBe(false);
+    });
+
+    it('should warn class in the locker', () => {
+        component.page.lockedByAnotherUser = true;
+        fixture.detectChanges();
+
+        const lockSwitch: DebugElement = de.query(By.css('.edit-page-toolbar__locker'));
+        expect(lockSwitch.classes.warn).toBe(true);
+    });
+
+    it('should have locker enabled', () => {
+        fixture.detectChanges();
+        const lockSwitch: DebugElement = de.query(By.css('.edit-page-toolbar__locker'));
+
+        expect(lockSwitch.componentInstance.disabled).toBe(false);
+    });
+
+    it('should have locker disabled', () => {
+        component.page.canLock = false;
+        fixture.detectChanges();
+        const lockSwitch: DebugElement = de.query(By.css('.edit-page-toolbar__locker'));
+
+        expect(lockSwitch.componentInstance.disabled).toBe(true);
+    });
+
+    it('should have page is locked message', () => {
+        component.page.canLock = false;
+        fixture.detectChanges();
+
+        const lockedMessage: DebugElement = de.query(By.css('.edit-page-toolbar__cant-lock-message'));
+        expect(lockedMessage.nativeElement.textContent).toContain('Page is locked');
+    });
+
+    it('should blink page is locked message', () => {
+        spyOn(component, 'lockerHandler');
+        component.page.canLock = false;
+        fixture.detectChanges();
+
+        const lockSwitch: DebugElement = de.query(By.css('.edit-page-toolbar__locker'));
+        lockSwitch.nativeElement.click();
```

Comment:
I dont understand this test, why are you clicking this element if canLock is false?

Reply:
Ok, so the switch is disabled but Jason and I think that people might click it because... people, so when people click it it will "blink" the message like saying: "Hey! stop doing that don't you see the page is locked?"